src/main/java/seedu/address/model/EventModelManager.java

```diff
@@ -0,0 +1,135 @@
+package seedu.address.model;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.nio.file.Path;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
+
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.model.event.Event;
+
+/**
+ * Represents the in-memory model of the address book data.
+ */
+public abstract class EventModelManager implements Model { //never use
+    private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
+
+    private final EventBook eventBook;
+    private final UserPrefs userPrefs;
+    private final FilteredList<Event> filteredEvents;
+
+    /**
+     * Initializes a ModelManager with the given addressBook and userPrefs.
+     */
+    public EventModelManager(ReadOnlyEventBook eventBook, ReadOnlyUserPrefs userPrefs) {
+        super();
+        requireAllNonNull(eventBook, userPrefs);
+
+        logger.fine("Initializing with address book: " + eventBook + " and user prefs " + userPrefs);
+
+        this.eventBook = new EventBook(eventBook);
+        this.userPrefs = new UserPrefs(userPrefs);
+        filteredEvents = new FilteredList<>(this.eventBook.getEventList());
+    }
+
+    public EventModelManager() {
+        this(new EventBook(), new UserPrefs());
+    }
+
+    //=========== UserPrefs ==================================================================================
+
+    @Override
+    public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+        requireNonNull(userPrefs);
+        this.userPrefs.resetData(userPrefs);
+    }
+
+    @Override
+    public ReadOnlyUserPrefs getUserPrefs() {
+        return userPrefs;
+    }
+
+    @Override
+    public GuiSettings getGuiSettings() {
+        return userPrefs.getGuiSettings();
+    }
+
+    @Override
+    public void setGuiSettings(GuiSettings guiSettings) {
+        requireNonNull(guiSettings);
+        userPrefs.setGuiSettings(guiSettings);
+    }
+
+    @Override
+    public Path getAddressBookFilePath() {
```

Comment:
getEventBookFilePath() and setEventBookFilePath()?

Reply:
Not using this class